File: app/src/main/java/com/cn29/aac/ui/shopping/AlbumFragmentModule.java
package com.cn29.aac.ui.shopping;

import com.cn29.aac.R;
import com.cn29.aac.databinding.FragmentAlbumBinding;
import com.cn29.aac.ui.shopping.vm.AlbumFragmentViewModel;
import com.cn29.aac.ui.shopping.vm.AlbumFragmentViewModelFactory;
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModel;

import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Charles Ng on 20/10/2017.
 */

@Module
public class AlbumFragmentModule {


  @Provides
  AlbumFragmentViewModel getArtistViewModel(AlbumFragmentViewModelFactory factory,
      AlbumFragment artistFragment) {
    return ViewModelProviders.of(artistFragment.getActivity(), factory)
        .get(AlbumFragmentViewModel.class);
  }

  @Provides
  ShoppingActivityViewModel getShoppingKartActivityViewModel(AlbumFragment albumFragment) {
    return ViewModelProviders.of(albumFragment.getActivity())
        .get(ShoppingActivityViewModel.class);
  }


  @Provides
  FragmentAlbumBinding provideBinding(AlbumFragment artistFragment) {
    return DataBindingUtil
        .inflate(artistFragment.getActivity().getLayoutInflater(), R.layout.fragment_album, null,
            false);
  }
}

File: app/src/main/java/com/cn29/aac/ui/shopping/AlbumFragmentModule.kt
package com.cn29.aac.ui.shopping

import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.cn29.aac.R
import com.cn29.aac.databinding.FragmentAlbumBinding
import com.cn29.aac.ui.shopping.vm.AlbumFragmentViewModel
import com.cn29.aac.ui.shopping.vm.AlbumFragmentViewModelFactory
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModel
import dagger.Module
import dagger.Provides

/**
 * Created by Charles Ng on 20/10/2017.
 */
@Module
class AlbumFragmentModule {
    @Provides
    fun getArtistViewModel(factory: AlbumFragmentViewModelFactory,
                           artistFragment: AlbumFragment): AlbumFragmentViewModel {
        return ViewModelProvider(artistFragment.requireActivity(), factory)
                .get(AlbumFragmentViewModel::class.java)
    }

    @Provides
    fun getShoppingKartActivityViewModel(albumFragment: AlbumFragment): ShoppingActivityViewModel {
        return ViewModelProvider(albumFragment.requireActivity())
                .get(ShoppingActivityViewModel::class.java)
    }

    @Provides
    fun provideBinding(artistFragment: AlbumFragment): FragmentAlbumBinding {
        return DataBindingUtil
                .inflate(artistFragment.requireActivity().layoutInflater,
                         R.layout.fragment_album,
                         null,
                         false)
    }
}